internal/pkg/file/delivery/delivery.go

```diff
@@ -0,0 +1,100 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package delivery
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/elastic/fleet-server/v7/internal/pkg/bulk"
+	"github.com/elastic/fleet-server/v7/internal/pkg/file"
+	"github.com/elastic/fleet-server/v7/internal/pkg/file/cbor"
+	"github.com/elastic/go-elasticsearch/v8"
+	"github.com/rs/zerolog"
+)
+
+var (
+	ErrNoFile = errors.New("file data not found")
+)
+
+type Deliverer struct {
+	sizeLimit int64
+
+	client *elasticsearch.Client
+	bulker bulk.Bulk
+}
+
+func New(client *elasticsearch.Client, bulker bulk.Bulk, sizeLimit int64) *Deliverer {
+	return &Deliverer{
+		client:    client,
+		bulker:    bulker,
+		sizeLimit: sizeLimit,
+	}
+}
+
+func (d *Deliverer) FindFileForAgent(ctx context.Context, fileID string, agentID string) (file.MetaDoc, error) {
+	result, err := findFileForAgent(ctx, d.bulker, fileID, agentID)
+	if err != nil {
+		return file.MetaDoc{}, err
+	}
+	if result == nil || len(result.Hits) == 0 {
+		return file.MetaDoc{}, ErrNoFile
+	}
+
+	var fi file.MetaDoc
+	if err := json.Unmarshal(result.Hits[0].Source, &fi); err != nil {
+		return file.MetaDoc{}, fmt.Errorf("file meta doc parsing error: %w", err)
+	}
+
+	return fi, nil
+}
+
+func (d *Deliverer) SendFile(ctx context.Context, zlog zerolog.Logger, w http.ResponseWriter, f file.MetaDoc, fileID string) error {
+
+	// find chunk indices behind alias, doc IDs
+	infos, err := file.GetChunkInfos(ctx, d.bulker, FileDataIndexPattern, fileID, file.GetChunkInfoOpt{})
+	if err != nil {
+		zlog.Error().Err(err).Msg("problem getting infos")
+		return err
+	}
+
+	if len(infos) == 0 {
+		zlog.Warn().Str("fileID", fileID).Msg("chunk documents not found for file")
+		w.WriteHeader(http.StatusNotFound)
```

Comment:
At this point the content headers from `internal/pkg/api/handleFileDelivery.go` have already been written, which means that the response code is (implicitly) set to 200. 
Is there a way to do this check earlier in `handleFileDelivery`?